src/methods/defining_methods/hard/q8/MainTest.java

```diff
@@ -0,0 +1,74 @@
+package methods.defining_methods.hard.q8;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.stream.Stream;
+
+import global.MethodTest;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import global.BaseTest;
+import global.tools.CustomAssertions;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.DoubleLiteral;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+
+public class MainTest extends BaseTest {
+    // Java
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "100";
+        return new Clause[]{
+                new StringLiteral("Enter the cost of rent per month:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new StringLiteral("The total cost of rent per year is \\$"),
+                new DoubleLiteral("rentYearly")
+        };
+    }
+
+    ;
+
+    public void runMain() {
+        RentMethod.main(new String[0]);
+    }
+
+    static Stream<Arguments> rentPricesProvider() {
+        return Stream.of(
+                Arguments.of(100.0, 1200.0),
+                Arguments.of(200.0, 2400.0),
+                Arguments.of(500.0, 6000.0),
```

Comment:
Also maybe use more double values with numbers other than zero in the decimal (to make sure they are not losing precision.